Formula/zim.rb

```diff
@@ -0,0 +1,47 @@
+class Zim < Formula
+  desc "Graphical text editor used to maintain a collection of wiki pages"
+  homepage "https://zim-wiki.org/"
+  url "https://github.com/jaap-karssenberg/zim-desktop-wiki/archive/0.72.1.tar.gz"
+  sha256 "ba02e418b4fb1d7847f96b49ada8c917c881a28bb5fb55dcdca54be7b3fd196a"
+  head "https://github.com/jaap-karssenberg/zim-desktop-wiki.git"
+
+  bottle do
+    cellar :any_skip_relocation
+    sha256 "91f15636c51398dc31f7c43d570a652cc1bddb8fc3571bdfc203d25231aa095d" => :catalina
+    sha256 "91f15636c51398dc31f7c43d570a652cc1bddb8fc3571bdfc203d25231aa095d" => :mojave
+    sha256 "91f15636c51398dc31f7c43d570a652cc1bddb8fc3571bdfc203d25231aa095d" => :high_sierra
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "adwaita-icon-theme"
+  depends_on "graphviz"
+  depends_on "gtk+3"
+  depends_on "gtksourceview3"
+  depends_on "pygobject3"
+  depends_on "python"
+
+  resource "pyxdg" do
+    url "https://files.pythonhosted.org/packages/47/6e/311d5f22e2b76381719b5d0c6e9dc39cd33999adae67db71d7279a6d70f4/pyxdg-0.26.tar.gz"
+    sha256 "fe2928d3f532ed32b39c32a482b54136fe766d19936afc96c8f00645f9da1a06"
+  end
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python3.7/site-packages"
+    resource("pyxdg").stage do
+      system "python", *Language::Python.setup_install_args(libexec/"vendor")
+    end
+    ENV["XDG_DATA_DIRS"] = libexec/"share"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python3.7/site-packages"
+    system "python3", "./setup.py", "install", "--prefix=#{libexec}"
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"], :XDG_DATA_DIRS => ["#{HOMEBREW_PREFIX}/share", libexec/"share"].join(":")
+    pkgshare.install "zim"
+  end
+
+  test do
+    ENV["LC_ALL"] = "en_US.UTF-8"
+    ENV["LANG"] = "en_US.UTF-8"
+
+    system "#{bin}/zim", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.